src/overlays/actors/ovl_Obj_Mure3/z_obj_mure3.c

```diff
@@ -31,41 +33,168 @@ const ActorInit Obj_Mure3_InitVars = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure3/func_8098F220.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure3/func_8098F364.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure3/func_8098F438.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure3/ObjMure3_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure3/ObjMure3_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure3/func_8098F598.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure3/func_8098F5AC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure3/func_8098F5D0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure3/func_8098F5E4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure3/func_8098F66C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure3/func_8098F680.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure3/ObjMure3_Update.s")
+void func_8098F040(ObjMure3* this, PlayState* play) {
+    s32 i;
+    Vec3f spawnPos;
+
+    Math_Vec3f_Copy(&spawnPos, &this->actor.world.pos);
+    for (i = 0; i < ARRAY_COUNT(this->unk148) - 1; i++, spawnPos.y += 20.0f) {
+        if (!((this->unk164 >> i) & 1)) {
+            this->unk148[i] = (EnItem00*)Item_DropCollectible2(play, &spawnPos, 0x10001);
+            if (this->unk148[i] != NULL) {
+                this->unk148[i]->actor.room = this->actor.room;
+            }
+        }
+    }
+}
+
+void func_8098F110(ObjMure3* this, PlayState* play) {
+    s32 i;
+    Vec3f spawnPos;
+    f32 sin = Math_SinS(this->actor.world.rot.y);
+    f32 cos = Math_CosS(this->actor.world.rot.y);
+    f32 radius;
+
+    spawnPos.y = this->actor.world.pos.y;
+
+    for (i = 0, radius = -40.0f; i < ARRAY_COUNT(this->unk148) - 1; i++, radius += 20.0f) {
+        if (!((this->unk164 >> i) & 1)) {
+            spawnPos.x = this->actor.world.pos.x + (sin * radius);
+            spawnPos.z = this->actor.world.pos.z + (cos * radius);
+            this->unk148[i] = (EnItem00*)Item_DropCollectible2(play, &spawnPos, 0x4000);
+            if (this->unk148[i] != NULL) {
+                this->unk148[i]->actor.room = this->actor.room;
+            }
+        }
+    }
+}
+
+void func_8098F220(ObjMure3* this, PlayState* play) {
+    s16 yRot;
+    Vec3f spawnPos;
+    s32 i;
+
+    spawnPos.y = this->actor.world.pos.y;
+    yRot = this->actor.world.rot.y;
+
+    for (i = 0; i < ARRAY_COUNT(this->unk148); i++) {
+        if (!((this->unk164 >> i) & 1)) {
+            spawnPos.x = (Math_SinS(yRot) * 40.0f) + this->actor.world.pos.x;
+            spawnPos.z = (Math_CosS(yRot) * 40.0f) + this->actor.world.pos.z;
+            this->unk148[i] = (EnItem00*)Item_DropCollectible2(play, &spawnPos, 0x4000);
+            if (this->unk148[i] != NULL) {
+                this->unk148[i]->actor.room = this->actor.room;
+            }
+        }
+        yRot += 0x2AAA;
```

Comment:
```suggestion
        yRot += 0x10000 / ARRAY_COUNT(this->unk148);
```
Though again, feels like it would be simpler to just have 6 here instead.